### PR TITLE
Adding metadata signing check

### PR DESCRIPTION
**Pull request by bcl**
Add support for checking repomd.xml gpg signature to sources.

This pull request includes:

- [X] adequate testing for the new functionality or fixed issue
- [X] adequate documentation informing people about the change such as
  - [X] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: dnf-json [hunk 1]
```diff
@@ -98,6 +99,9 @@ class Solver():
 
             for key in desc["gpgkeys"]:
                 if key.startswith("-----BEGIN PGP PUBLIC KEY BLOCK-----"):
+                    # Not using with because it needs to be a valid file for the duration. It
+                    # is inside the temporary persistdir so will be cleaned up on exit.
+                    # pylint: disable=consider-using-with
                     keyfile = tempfile.NamedTemporaryFile(dir=keydir, delete=False)
```
Comment (croissanne) [inline dnf-json:105]: As long as `delete=False` you can still use `with...` I think? The file won't be removed on close.

File: dnf-json [hunk 2]
```diff
@@ -82,6 +82,29 @@ class Solver():
         if "sslclientcert" in desc:
             repo.sslclientcert = desc["sslclientcert"]
 
+        if "check_gpg" in desc:
+            repo.gpgcheck = desc["check_gpg"]
+        if "check_repogpg" in desc:
+            repo.repo_gpgcheck = desc["check_repogpg"]
+        if "gpgkey" in desc:
```
Comment (croissanne) [inline dnf-json:90]: Is this kept for compatibility reasons?

File: cmd/osbuild-composer-cli-tests/main_test.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build integration
```
Comment (croissanne) [inline cmd/osbuild-composer-cli-tests/main_test.go:1]: Is the `// +build integration` below it not sufficient?

**Review by croissanne — COMMENTED** [review]
thank you! lgtm overall

File: dnf-json (hunk 1 above)
Comment (bcl) [reply dnf-json:105]: Possibly, but I think this is clearer to future me :)

File: dnf-json (hunk 2 above)
Comment (bcl) [reply dnf-json:90]: correct, no need to break dnf-json's api.

File: cmd/osbuild-composer-cli-tests/main_test.go (hunk 3 above)
Comment (bcl) [reply cmd/osbuild-composer-cli-tests/main_test.go:1]: Interesting. I didn't add that on purpose. I'm guessing vim-go may have done that...

Comment (bcl) [reply cmd/osbuild-composer-cli-tests/main_test.go:1]: Ah, I see. go:build is now preferred by go 1.18 and, so I'll remove the old form.

**Review by croissanne — APPROVED** [review]
Lovely!

**Pull request merged**